File: Leetcode/1200. Minimum Absolute Difference.ts
function minimumAbsDifference(arr: number[]): number[][] {
    arr.sort((a,b) => a-b);
    let m = new Map<number, number[][]>();
    
    for(let i=1; i<arr.length; i++) {
        let d: number = arr[i] - arr[i-1];
        let a: number[][] = m.get(d) ?? [];
        a.push([arr[i-1], arr[i]])
        m.set(d, a);
    }
    
    let s: number = Number.MAX_SAFE_INTEGER;
    
    for(let [key, value] of m.entries()) {
        if(key < s) s = key;
    }
    
    return m.get(s);
};

File: Leetcode/1200. Minimum Absolute Difference.rs
use std::cmp;
use std::collections::*;
impl Solution {
    pub fn minimum_abs_difference(arr: Vec<i32>) -> Vec<Vec<i32>> {
        let mut arr2 = arr.to_vec();
        arr2.sort();
        
        let mut map:BTreeMap<i32,Vec<Vec<i32>>> = BTreeMap::new();
        for i in 1..arr2.len() {
            let key = arr2[i]-arr2[i-1];
            let mut val = [arr2[i-1], arr2[i]].to_vec();
            
            match map.get_mut(&key) {
                Some(temp) => {
                    temp.push(val);
                },
                None => {
                    map.insert(key, vec![val]);
                }
            }
        }
        
        map.values().next().unwrap().to_vec()
    }
}
